### PR TITLE
feature/service addons

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Allow additional services to be plugged into Packetfence

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add OpenAPI specification
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Allow other services to be plugged into packetfence

### CODE REVIEW

File: lib/pf/dal/_node_meta.pm [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+our @COLUMN_NAMES;
+
+BEGIN {
+    @FIELD_NAMES = qw(
+        name
+        mac
+        value
+        created_at
+        updated_at
+        expired_at
```
Comment (satkunas) [inline lib/pf/dal/_node_meta.pm]: remove

File: lib/pf/dal/_node_meta.pm [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    );
+
+    %DEFAULTS = (
+        name => '',
+        mac => '',
+        value => undef,
+        expired_at => '0000-00-00 00:00:00',
```
Comment (satkunas) [inline lib/pf/dal/_node_meta.pm]: remove

File: lib/pf/dal/_node_meta.pm [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            is_auto_increment => 0,
+            is_primary_key => 0,
+            is_nullable => 1,
+        },
+        expired_at => {
+            type => 'DATETIME',
+            is_auto_increment => 0,
+            is_primary_key => 0,
+            is_nullable => 1,
+        },
```
Comment (satkunas) [inline lib/pf/dal/_node_meta.pm]: remove

File: lib/pf/dal/_node_meta.pm [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        mac
+    );
+
+    @COLUMN_NAMES = qw(
+        node_meta.name
+        node_meta.mac
+        node_meta.value
+        node_meta.created_at
+        node_meta.updated_at
+        node_meta.expired_at
```
Comment (satkunas) [inline lib/pf/dal/_node_meta.pm]: remove

**Review by satkunas — CHANGES_REQUESTED** [review]

File: lib/pf/dal/_node_meta.pm (hunk 4 above)
Comment (jrouzierinverse) [reply lib/pf/dal/_node_meta.pm]: All fixed

**Pull request merged**